pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -163,6 +167,12 @@ Future<PackageSummary> summarizePackageArchive(String archivePath) async {
     examplePath = null;
   }
 
+  String licensePath = searchForFile(licenseFileNames);
+  final licenseContent = await extractContent(licensePath);
+  if (licenseContent == null) {
+    licensePath = null;
+  }
```

Comment:
I think we should be okay to require that new packages have a `LICENSE` file and that the name of the file is `LICENSE`.

Ideally, it would be nice if when we read old packages we maintain the more relaxed rules we've previously had.

Reply:
This code is also used by the package asset backfill, and will be applied on old versions.

Reply:
okay, I guess making what we accept more strict is something we can do later :D